wal/src/message_queue_impl/region_meta.rs

```diff
@@ -0,0 +1,606 @@
+#[allow(unused)]
+impl RegionMeta {
+    // TODO: Need to implement the init method using the [RegionMetaSnapshot] which
+    // will be persisted.
+
+    pub async fn prepare_for_table_write(&self, table_id: TableId) -> SequenceNumber {
+        {
+            let inner = self.inner.read().await;
+            if let Some(table_meta) = inner.table_metas.get(&table_id) {
+                return table_meta.get_meta_data().await.next_sequence_num;
+            }
+        }
+
+        // Double check is not needed, due to just one task will write the specific
+        // table.
+        let mut inner = self.inner.write().await;
+        assert!(inner
+            .table_metas
+            .insert(table_id, TableMeta::new(table_id))
+            .is_none(),
+            "now just support the thread model: one writer to one table, make no sense to occur race here");
+        // New table, so returned next sequence num is zero.
+        SequenceNumber::MIN
+    }
+
+    /// Update following meta data of table after each writing:
+    /// + mapping of start sequence number to start offset
+    /// + high watermark
+    /// + next sequence number
+    pub async fn update_after_table_write(
+        &self,
+        table_id: TableId,
+        write_offset_range: OffsetRange,
+    ) -> Result<()> {
+        ensure!(
+            write_offset_range.start <= write_offset_range.end,
+            UpdateAfterWrite {
+                table_id, msg: format!("write offset range's start should not be larger than its end, offset range:{:?}", 
+                write_offset_range)
+            },
+        );
+
+        let inner = self.inner.read().await;
+        let table_meta = inner
+            .table_metas
+            .get(&table_id)
+            .with_context(|| UpdateAfterWrite {
+                table_id,
+                msg: format!(
+                    "table:{}'s meta not found while update after its write",
+                    table_id
+                ),
+            })?;
+
+        table_meta.update_after_write(write_offset_range).await;
+
+        Ok(())
+    }
+
+    /// Mark the deleted sequence number to latest next sequence number.
+    pub async fn mark_table_deleted(
+        &self,
+        table_id: TableId,
+        sequence_num: SequenceNumber,
+    ) -> Result<()> {
+        let inner = self.inner.read().await;
+        let table_meta = inner
+            .table_metas
+            .get(&table_id)
+            .with_context(|| MarkDeleted {
+                table_id,
+                msg: format!("table:{}'s meta not found while mark its deleted", table_id),
+            })?;
+        table_meta.mark_deleted(sequence_num).await;
+
+        Ok(())
+    }
+
+    /// Scan the table meta entry in it and get the snapshot about tables' next
+    /// sequences.
+    ///
+    /// NOTICE: Need to freeze the whole region meta on high-level before
+    /// calling.
+    pub async fn get_snapshot(&self) -> RegionMetaSnapshot {
```

Comment:
How about replacing `get` with `making` because it's not a trivial thing to build the snapshot: 
```suggestion
    pub async fn make_snapshot(&self) -> RegionMetaSnapshot {
```

Reply:
Done.